mail_tracking/models/mail_message.py

```diff
@@ -63,35 +98,51 @@ def tracking_status(self):
             partners -= partners_already
             for partner in partners:
                 # If there is partners not included, then status is 'unknown'
-                partner_trackings.append((
-                    'unknown', False, partner.name, partner.id, partner.email))
+                # Because can be an Cc recipient
+                isCc = False
+                if partner.email in email_cc_list:
+                    email_cc_list.discard(partner.email)
+                    isCc = True
+                partner_trackings.append({
+                    'status': 'unknown',
+                    'status_human':
+                        self._partner_tracking_status_human_get('unknown'),
+                    'error_type': False,
+                    'error_description': False,
+                    'tracking_id': False,
+                    'recipient': partner.name,
+                    'partner_id': partner.id,
+                    'isCc': isCc,
+                })
+            for email in email_cc_list:
+                # If there is Cc without partner
+                partner_trackings.append({
+                    'status': 'unknown',
+                    'status_human':
+                        self._partner_tracking_status_human_get('unknown'),
+                    'error_type': False,
+                    'error_description': False,
+                    'tracking_id': False,
+                    'recipient': email,
+                    'partner_id': False,
+                    'isCc': True,
+                })
             res[message.id] = partner_trackings
         return res
 
     @api.multi
-    def _get_email_cc(self):
-        """This method gets all Cc mails and the associated partner if exist.
-            The result is a dictionary by 'message id' with a list of tuples
-            (str:email_cc, list:[partner id, partner display_name] or False)
-        """
+    def _get_failed_message(self):
         res = {}
-        ResPartnerObj = self.env['res.partner']
         for message in self:
-            email_cc_list = email_split(message.email_cc)
-            email_cc_list_checked = []
-            if any(email_cc_list):
-                partners = ResPartnerObj.search([
-                    ('email', 'in', email_cc_list)
-                ])
-                email_cc_list = set(email_cc_list)
-                for partner in partners:
-                    email_cc_list.discard(partner.email)
-                    email_cc_list_checked.append(
-                        (partner.email, [partner.id, partner.display_name]))
-                for email in email_cc_list:
-                    email_cc_list_checked.append((email, False))
             res.update({
-                message.id: email_cc_list_checked
+                message.id: message.mail_tracking_needs_action
```

Comment:
As a suggestion, improving code readability will help here. Code is written once but read many times, so write it as obvious as possible. Example:

```python
    def _get_failed_message(self):
        result = {}
        failed_states = self.get_failed_states()
        for message in self:
            needs_action = message.mail_tracking_needs_action
            involves_me = self.env.user.partner_id in (message.author_id | message.partner_ids)
            has_failed_trackings = failed_states.intersection(message.mapped("state"))
            result[message.id] = bool(needs_action and involves_me and has_failed_trackings)
```

OTOH, have you considered transforming this into a computed non-stored field? IMHO that'd be easier to use than this method that returns a dict to be later inspected by id.